data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -83,27 +83,47 @@ public static List<DataFile> listPartition(
     return listPartition(partition, uri, format, spec, conf, metricsConfig, mapping, 1);
   }
 
+  /**
+   * Returns the data files in a partition by listing the partition location using some number of
+   * threads.
+   *
+   * <p>For Parquet and ORC partitions, this will read metrics from the file footer. For Avro
+   * partitions, metrics are set to null.
+   *
+   * <p>Note: certain metrics, like NaN counts, that are only supported by Iceberg file writers but
+   * not file footers, will not be populated.
+   *
+   * @param partition map of partition columns to column values
+   * @param uri partition location URI
+   * @param format partition format, avro, parquet or orc
+   * @param spec a partition spec
+   * @param conf a Hadoop conf
+   * @param metricsConfig a metrics conf
+   * @param mapping a name mapping
+   * @param parallelism number of threads to use
+   * @return a List of DataFile
+   */
   public static List<DataFile> listPartition(
-      Map<String, String> partitionPath,
-      String partitionUri,
+      Map<String, String> partition,
```

Comment:
partitions ? minor

Reply:
I'm confused about what happened here, are we modifying the exiting public method or adding a new one? We should attempt not to break compatibility if we can for public methods in utility classes.

Reply:
We are modifying an existing public method. There is no change to the signature of the method, although the names of a few parameters were changed to make them consistent with the other existing `listPartitions` method (which was actually the first `listPartitions` method). The difference between the first `listPartitions` and this second one is that this has an additional `parallelism` parameter. The javadoc for the first method is copied over to this second method with relevant addition (another reason to make the parameter names consistent).